### PR TITLE
Create hosted machine

### PR DESCRIPTION
**Pull request by jameshaydon**

### CODE REVIEW

File: rad/prelude/http.rad [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ :exports '[json-get! json-post!]}
+
+(import prelude/strings ['intercalate] :unqualified)
+(import prelude/io :unqualified)
+
+(def json-req!
+  (fn [url extra-opts]
+    (def opts (<> ["-s" url "-H" "Accept: application/json"] extra-opts))
+    (match (process-with-stdout-stderr-exitcode! "curl" opts "")
+           ['resp _ :ok]  (from-json (intercalate "" resp))
```
[the remark this answers is not in the document]
Comment (jkarni) [reply rad/prelude/http.rad]: Also, `unwords`

File: rad/prelude/http.rad [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ :doc "Functions for making HTTP requests."
+ :exports '[json-get! json-post!]}
+
+(import prelude/strings ['intercalate] :unqualified)
+(import prelude/io :unqualified)
+
+(def json-req!
+  (fn [url extra-opts]
+    (def opts (<> ["-s" url "-H" "Accept: application/json"] extra-opts))
+    (match (process-with-stdout-stderr-exitcode! "curl" opts "")
```
Comment (jkarni) [inline rad/prelude/http.rad]: Could also be a primfn, no?

File: rad/prelude/hosted-machine.rad [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{:module 'prelude/hosted-machine
+ :doc "Functions for interacting with hosted machines."
+ :exports '[]}
```
Comment (jkarni) [inline rad/prelude/hosted-machine.rad]: I imagine `new-hosted-machine!` should be exported?

File: rad/prelude/http.rad (hunk 1 above)
[the remark this answers is not in the document]
Comment (jameshaydon) [reply rad/prelude/http.rad]: It's really `unlines`, i'll at it to `prelude/strings`.

**Pull request closed without merging**